### PR TITLE
Refactor modularize code

### PR DESCRIPTION
**Pull request by kripken**
Refactor emcc.py to more cleanly use 'final' as the current updating js, and assign it to js_target at the very end, never to modify it again. With that in place, move modularize to right before that, so that it can work with things like closure+wasm (if we modularize first, closure is confused).

This fixes box2d.js and ammo.js with closure+wasm.

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -2224,7 +2204,29 @@ def do_minify(): # minifies the code. this is also when we do certain optimizati
               assert JSOptimizer.cleanup_shell
               logging.debug('running cleanup on shell code')
               final = shared.Building.js_optimizer_no_asmjs(final, ['noPrintMetadata', 'JSDCE', 'last'] + (['minifyWhitespace'] if JSOptimizer.minify_whitespace else []))
-            shutil.move(final, js_target)
+            if DEBUG: save_intermediate('postclean', 'js') # waka
+
+      if shared.Settings.MODULARIZE: # waka
```
Comment (jgravelle-google) [inline emcc.py]: Extra `waka`

File: emcc.py [hunk 2]
```diff
@@ -2224,7 +2204,29 @@ def do_minify(): # minifies the code. this is also when we do certain optimizati
               assert JSOptimizer.cleanup_shell
               logging.debug('running cleanup on shell code')
               final = shared.Building.js_optimizer_no_asmjs(final, ['noPrintMetadata', 'JSDCE', 'last'] + (['minifyWhitespace'] if JSOptimizer.minify_whitespace else []))
-            shutil.move(final, js_target)
+            if DEBUG: save_intermediate('postclean', 'js') # waka
```
Comment (jgravelle-google) [inline emcc.py]: Extra `waka`

File: emcc.py [hunk 3]
```diff
@@ -2224,7 +2204,29 @@ def do_minify(): # minifies the code. this is also when we do certain optimizati
...
+        f.write('  ' + shared.Settings.EXPORT_NAME + ' = ' + shared.Settings.EXPORT_NAME + ' || {};\n')
+        f.write('  var Module = ' + shared.Settings.EXPORT_NAME + ';\n') # included code may refer to Module (e.g. from file packager), so alias it
+        f.write('\n')
+        f.write(src)
+        f.write('\n')
+        f.write('  return ' + shared.Settings.EXPORT_NAME + ';\n')
+        f.write('};\n')
+        f.close()
+        src = None
+        if DEBUG: save_intermediate('modularized', 'js') # waka
```
Comment (jgravelle-google) [inline emcc.py]: Extra `waka`

**Comment by kripken** [comment]
Heh, I add those `waka`s when I want to repeatedly cycle through locations in code when refactoring. I usually remember to remove them... ;)

**Comment by curiousdannii** [comment]
Any chance you could fix #5093 at the same time? :)

**Comment by kripken** [comment]
Sorry, I don't know what the right fix is for that other issue.

**Pull request merged**